types/recoil/lib/core/RecoilRoot.d.ts

```diff
@@ -0,0 +1,11 @@
+import { FC } from 'react';
```

Comment:
These additional files do not exist in the package so shouldnt be added here
https://unpkg.com/browse/recoil@0.0.7/

Reply:
@sheetalkamat I added `///<reference types="react" />` to the top of the file in the latest commit-- does that solve the issue? I'm ultimately trying to express that the export of this file is a React function component

Reply:
This is a fairly long and somewhat confusing thread, so I'm not sure I'm entirely caught up, but I think @sheetalkamat may have been asking why there are bunch of .d.ts files in lib/core without corresponding .js files in the distributed code.  Maybe the concern was that they might create the impression that `require('recoil/lib/core/node')` is valid even though it won't resolve at runtime?

Reply:
ah I see, that makes sense--thanks for clarifying @amcasey! If we absolutely have to we can put everything in one file, but I'd really like to keep them in separate files for maintainability's sake. The intention was that the `lib` folder here would be private and not accessed directly. Is there a standard name we should use other than `lib` to denote a private folder @sheetalkamat?

Reply:
To keep things moving I consolidated everything into one file (thanks to @mathieucaroff for having done that in a separate comment). Marked as resolved